doc/designs/prci_checker.md

```diff
@@ -0,0 +1,150 @@
+  - nightly_ipa-4-10_previous
+
+# Info specific to prci definition files
+# 'job_prefix' field is mandatory
+# 'update_packages', 'selinux_enforcing', 'enable_testing_repo' and
+# 'copr_defined' are optional boolean fields (if not specified,
+# false value is assumed).
+# New definitions specifications may be added anytime
+prci_job_spec:
+  gating:
+    job_prefix: 'fedora-latest-ipa-4-10/'
+  nightly_ipa-4-10_latest:
+    job_prefix: 'fedora-latest-ipa-4-10/'
+  nightly_ipa-4-10_latest_selinux:
+    job_prefix: 'fedora-latest-ipa-4-10/'
+    selinux_enforcing: True
+  nightly_ipa-4-10_previous:
+    job_prefix: 'fedora-previous-ipa-4-10/'
+```
+
+## Error cases examples
+
+After adding new jobs to the definition or editing the content, it is common that a developer has made some mistakes following the convention. This checker will help us correcting these errors. Examples:
+
+###   Wrong or unsupported class:
+
+``` bash
+$ python3 ipatests/prci_definitions/prci_checker.py -f ipatests/prci_definitions/gating.yaml
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] gating [Jobs] 24
+ERROR: In job 'fedora-latest/test_installation_TestInstallMaster':
+  'class' field should be defined with one of the supported: ['RunPytest', 'RunPytest2', 'RunPytest3', 'RunWebuiTests', 'RunADTests']
+CHECKS FINISHED WITH ERRORS
+```
+
+### Wrong job prefix in gating:
+
+``` bash
+$ python3 ipatests/prci_definitions/prci_checker.py -f ipatests/prci_definitions/gating.yaml
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] gating [Jobs] 24
+ERROR: In job 'fedora-latdest/test_installation_TestInstallMaster':
+  Job name should start with prefix 'fedora-latest/'
+CHECKS FINISHED WITH ERRORS
+```
+
+### Wrong or missing parameter values.
+E.g. missing `selinux_enforcing: True` in selinux job definition:
+
+``` bash
+$ python3 ipatests/prci_definitions/prci_checker.py -f ipatests/prci_definitions/nightly_latest_selinux.yaml
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] nightly_latest_selinux [Jobs] 149
+ERROR: In job 'fedora-latest/simple_replication':
+  Job field "selinux_enforcing" should be defined as: "selinux_enforcing: True"
+CHECKS FINISHED WITH ERRORS
+```
+
+### Uneven number of jobs between definition files defined by spec file:
+
+``` bash
+$ python3 ipatests/prci_definitions/prci_checker.py
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] nightly_latest_pki [Jobs] 75
+[File] nightly_latest [Jobs] 148
+[File] nightly_latest_testing [Jobs] 149
+[File] nightly_rawhide [Jobs] 149
+[File] gating [Jobs] 24
+[File] nightly_latest_selinux [Jobs] 149
+[File] nightly_latest_testing_selinux [Jobs] 149
+[File] nightly_latest_sssd [Jobs] 16
+[File] nightly_latest_389ds [Jobs] 45
+[File] nightly_previous [Jobs] 149
+ERROR: Following PRCI definitions should have the same number of jobs: ['nightly_latest', 'nightly_latest_testing', 'nightly_rawhide', 'nightly_latest_selinux', 'nightly_latest_testing_selinux', 'nightly_previous']
+CHECKS FINISHED WITH ERRORS
+```
```

Comment:
I would add more details, for instance by taking an example from the prci_jobs_spec.yaml file and explaining what it means.
```
  nightly_latest_389ds:
    job_prefix: '389ds-fedora/'
    update_packages: True
    copr_defined: True
```
All the test definitions in the file nightly_latest_389ds.yaml must follow these rules:
- each job must have a name starting with 389ds-fedora/
For instance the following is ok:
```
 389ds-fedora/simple_replication:
    requires: [389ds-fedora/build]
    priority: 50
    job:
      class: RunPytest
      args:
        build_url: '{389ds-fedora/build_url}'
        update_packages: True
        copr: '@389ds/389-ds-base-nightly'
        test_suite: test_integration/test_simple_replication.py
        template: *ci-master-latest
        timeout: 3600
        topology: *master_1repl
```
- each job defined in the file must define `update_packages: True` and `copr_defined: True`. Since the file is testing with a copr repo for 389ds, the test must call `dnf update` in order to update the package to the version provided in the copr repo. Otherwise they would run with a wrong version of 389ds and would not properly test the integration. Likewise, it they define update_packages but do not set any copr repo, the wrong packages would be used.

Reply:
Thanks for the suggestions. Example added with some extra information